draft-ietf-ppm-dap.md

```diff
@@ -1305,50 +1493,89 @@ given batch interval, denoted `batch_interval`, decryption works as follows:
 
 ~~~
 agg_share = OpenBase(enc_share.enc, sk, "dap-01 aggregate share" ||
-    server_role || 0x00, task_id || batch_interval, enc_share.payload)
+    server_role || 0x00, task_id || batch_selector, enc_share.payload)
 ~~~
 
 where `sk` is the HPKE secret key, `task_id` is the task ID for the collect
 request, and `server_role` is the role of the server that sent the aggregate
-share (`0x02` for the leader and `0x03` for the helper).
+share (`0x02` for the leader and `0x03` for the helper). The value of
+`batch_selector` is computed by the Collector from its query and the response to
+its query:
+
+* For time-interval tasks, the batch selector is the batch interval specified in
+  the query.
 
-### Validating Batch Parameters {#batch-parameter-validation}
+* For fixed-size tasks, the batch selector is the batch ID assigned sent in the
+  response.
+
+### Batch Validation {#batch-validation}
```

Comment:
Can we refactor this such that there are subsections for each type of query? I know that there's some shared logic, but the mixed time/fixed-size validation steps is a bit hard to parse, especially if I'm only implementing one of them.

Reply:
Good idea, I will work on this now.

Reply:
Done, please have a look.

Reply:
Looks good!